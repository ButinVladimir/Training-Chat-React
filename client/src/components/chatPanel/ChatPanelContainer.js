import { connect } from 'react-redux';
import ChatPanel from './ChatPanel';

const mapStateToProps = state => ({
  showError: state.errorPanel.showError,
});

const ChatPanelContainer = connect(mapStateToProps)(ChatPanel);

export default ChatPanelContainer;
