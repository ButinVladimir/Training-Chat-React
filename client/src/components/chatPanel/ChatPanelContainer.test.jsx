import React from 'react';
import { Socket } from 'socket.io-client';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import ChatPanelContainer from './ChatPanelContainer';

describe('ChatPanelContainer', () => {
  let mockStore;
  let mockSocket;

  beforeEach(() => {
    mockStore = configureStore([]);
    mockSocket = new Socket();
  });

  it('renders chat panel', () => {
    const store = mockStore({
      errorPanel: {
        showError: false,
        error: '',
      },
      chatPanel: {
        messages: [],
      },
    });
    const component = mount(
      <Provider store={store}>
        <ChatPanelContainer socket={mockSocket} />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });

  it('renders error panel', () => {
    const store = mockStore({
      errorPanel: {
        showError: true,
        error: 'An error',
      },
      chatPanel: {
        messages: [],
      },
    });
    const component = mount(
      <Provider store={store}>
        <ChatPanelContainer socket={mockSocket} />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
