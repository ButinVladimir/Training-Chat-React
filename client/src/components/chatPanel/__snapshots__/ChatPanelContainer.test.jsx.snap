// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatPanelContainer renders chat panel 1`] = `
<div
  className="chat-messages-list-container"
>
  <ul
    className="chat-messages-list"
  />
</div>
`;

exports[`ChatPanelContainer renders error panel 1`] = `
<div
  className="error-panel-container"
>
  <div
    className="error-panel"
  >
    <div
      className="popup-row"
    >
      An error occurred
    </div>
    <div
      className="popup-row"
    >
      An error
    </div>
    <div
      className="popup-row"
    >
      <button
        className="close-button"
        onClick={[Function]}
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
